src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,243 @@
+      searchJointSets(jointSets.tail, inputVolume, numProcessors)
+    }
+  }
+
+  /**
+    * Cycles through persistent joint set to find joints that approximately balances load across processes
+    *
+    * @param jointSet Seq of Joints representing a persistent joint set
+    * @param rockVolume Block representing input rock volume that should be divided into approximately equal pieces
+    *                   for each process to operate on
+    * @param numProcessors Number of processes that will be used in analysis
+    * @param totalVolume Total volume of rockVolume
+    * @param tolerance Tolerance in deviation from ideal volume per processor. Default initial value is 1%, though it
+    *                  will be updated as necessary in recursive calls.
+    * @param stepSize Step size that tolerance will be incremented by if unable to find enough seed joints. Default
+    *                 initial value is 1%, though it will be updated as necessary in recursive calls.
+    * @return If able to find the required number of seed joints, this function will return a Seq of Joints. If unable
+    *         to find sufficient seed joints, it will return None.
+    */
+  @tailrec
+  private def findSeedJoints(jointSet: Seq[Joint], rockVolume: Block, numProcessors: Integer, totalVolume: Double,
+                     tolerance: Double = 0.01, stepSize: Double = 0.01): Option[Seq[Joint]] = {
+    if (jointSet.length < numProcessors - 1) {
+      println(s"Error: Not enough joints to generate required number of seed joints. ${jointSet.length} joints but"+
+      s" $numProcessors processors. Try using less processors.")
+      return None
+    }
+    val volumePerProc = rockVolume.volume / numProcessors
+    val seedJoints = cycleJointSet(jointSet, rockVolume, volumePerProc, totalVolume,
+      Seq.empty[Joint], tolerance, numProcessors)
+
+    if (stepSize < 0.01) {
+      println("Error: Unable to find satisfactory seed joints using current joint set. Please try another " +
+        "persistent joint set.")
+      None
+    } else if (seedJoints.length != numProcessors - 1) {
+      val newStepSize = if (tolerance >= 1.0) { stepSize / 2.0 } else { stepSize }
+      val newTolerance = if (tolerance >= 1.0) { newStepSize } else { tolerance + stepSize }
+      println(s"Warning: Could not find the required ${numProcessors - 1}, could only find ${seedJoints.length}")
+      println(s"Re-running with lower tolerance: %.1f%%".format(newTolerance * 100.0))
+      findSeedJoints(jointSet, rockVolume, numProcessors, totalVolume, newTolerance, newStepSize)
+    } else {
+      val initialBlock = Seq(rockVolume)
+      val blocks = seedJoints.foldLeft(initialBlock) { (currentBlocks, joint) =>
+        currentBlocks.flatMap(_.cut(joint))
+      }
+      val nonRedundantBlocks = blocks map { case block @ Block(blockCenter, _, _) =>
+          Block(blockCenter, block.nonRedundantFaces)
+      }
+      val volumes = nonRedundantBlocks.map(_.volume)
```

Comment:
I think it depends on how expensive the volume computations are. If they're pretty involved, we could consider using a lazy val. If it involves an LP stuff I'd say it could be worth it.

But this could all be premature optimization anyways.